packages/offix-datastore/src/replication/GraphQLCrudQueryBuilder.ts

```diff
@@ -0,0 +1,49 @@
+import gql from "graphql-tag";
+
+import { GraphQLQueryBuilder, GraphQLQueries } from "./GraphQLReplicator";
+import { Model } from "../Model";
+
+/**
+ * Builds GraphQL queries for models following
+ * the [GraphQLCRUD specification]{@link https://graphqlcrud.org/}
+ */
+export class GraphQLCrudQueryBuilder implements GraphQLQueryBuilder {
+    build(models: Model<any>[]): Map<string, GraphQLQueries> {
+        const queriesMap: Map<string, GraphQLQueries> = new Map();
+
+        models.forEach((model) => {
+            const fields = model.getFields();
+            const fieldsBuilder: string[] = Object.keys(fields).map((key) => {
+                const graphQLKey = fields[key].key;
+                return graphQLKey;
+            });
+            const graphQLFields = fieldsBuilder.join("\n");
+            const modelName = model.getName();
+
+            const mutations = {
+                create: gql`
+                mutation create${modelName}($input: Create${modelName}Input!) {
```

Comment:
@Eunovo @wtrocki we should probably have some documentation to explain that this is aligned to the graphqlcrud spec.

Reply:
I would use this as a prototype for the moment. We winning and getting lots of progress. Once we get prototype working end to end with all we want to have, we will need to smooth it up and document it

Reply:
Sure, for now, I only add a jsDoc comment on the class to indicate it's purpose